lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1781,18 +1767,13 @@ static bool performCompileStepsPostSILGen(CompilerInstance &Instance,
     return processCommandLineAndRunImmediately(
         Instance, std::move(SM), MSF, observer, ReturnValue);
 
-  llvm::StringSet<> LinkerDirectives;
-  collectLinkerDirectives(Invocation, MSF, LinkerDirectives);
-  // Don't proceed to IRGen if collecting linker directives failed.
-  if (Context.hadError())
```

Comment:
I take it we don't have tests for this failure path. We should get this to fire with a malformed TBD file.

Reply:
We have a failure path in IRGen that bails if an error is raised during emission, which should catch this case. We may do a bit of extra unnecessary IRGen compared to this early return, but I didn't think the TBDGen error path would be one that's hit all that often.

Reply:
Yeah, really only mastering errors would cause something like this.

Reply:
Looks like the test `original-defined-attr-linker-directives-fail.swift` exercises the error path, so marking this resolved.